tests/database_test_utils.py

```diff
@@ -0,0 +1,113 @@
+            default_specification["name"] = "Status " + str(specification["id"])
+            default_specification["sort_order"] = specification["id"]
+        return default_specification | specification
+
+    def _fix_missing_relations(self, specification: dict[str, Any]) -> None:
+        """
+        If a dictionary with a known table is given, like a task, the related item,
+        for example a status, will be created in the database if the id does not
+        exist yet. We do this to comply with database relationships and make it
+        easier to set up tests with minimal effort.
+        :param specification: a dictionary with a table specification
+        :return: None
+        """
+        if specification["table"] == "task":
+            status_specification = {"id": specification["status_id"], "table": "status"}
+            if not self.item_exists(status_specification):
+                self.init([status_specification])
+
+    def get_items(self, specification: dict[str, str | int]) -> Any:
+        """
+        Create a query based on the dictionary specification and return the result
+        :param specification: a dictionary with a table specification
+        :return: the results of the query
+        """
+        values = ", ".join(
+            key + "=" + str(val) if str(val).isnumeric() else str('"' + val + '"')  # type: ignore
+            for key, val in specification.items()  # type: ignore
+            if key != "table"  # type: ignore
+        )
+        table = specification["table"]
+        statement = f"SELECT * FROM {table} WHERE {values}"  # noqa S608
+        with Session(get_engine()) as session:
+            return session.exec(text(statement)).all()  # type: ignore
+
+    def item_exists(self, specification: dict[str, Any]) -> bool:
+        """
+        Check if an item exists in the database with the table and id given
+        in the dictionary
+        :param specification: a dictionary with a table specification
+        :return: True if the item exists in the database, False otherwise
+        """
+        result = self.get_items(specification)
+        return len(result) != 0
+
+    def init(self, specifications: list[dict[str, str | int]] | None = None) -> None:
+        """
+        Given an array of specifications, create the database entries.
+
+        Example: [{'table': 'task', 'id': 1 'title': 'Test task 1', 'description': 'Test task description 1'}]
+
+        The example below will be enriched so all required fields for a task will have a value.
+
+        Example: [{'table': 'task', 'id': 1}]
+
+        Example: [{"table": "status", "id": 1},{"table": "task", "id": 1, "status_id": 1}]
+
+        :param specifications: an array of dictionaries with table specifications
+        :return: None
+        """
+        if specifications is None:
+            return
+        for specification in specifications:
+            specification = self._enrich_with_default_values(specification)
```

Comment:
i would not add this, because the functions add fields that are not intended by the test writer

Reply:
There are quite some fields required by the database which are needed for a test. So you can create a simple task using f.e. only and id, and get default values. It does exactly what I want. If it is a good idea, not sure, we can discuss that.